src/timeoffset.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{Epoch, TimeScale, Unit};
+
+/// Errors during [TimeOffset] exploitation
+#[derive(Debug)]
+pub enum TimeOffsetError {
+    /// Timescales should not be identical when defining a new [TimeOffset]!
+    IdenticalTimescales,
+    /// Epoch is not expressed in one of the two supported [TimeScale]s!
+    NotSupportedTimescale,
+    /// Epoch should be expressed in the left-hand side [TimeScale].
+    InvalidTimescale,
+    /// [TimeOffset] is outdated and should not apply (difference is too large): should be updated weekly at least!
+    OutdatedTimeOffset,
+}
+
+/// [TimeOffset] used in [TimeShift]ing operations
+#[derive(Copy, Clone, PartialEq)]
+pub struct TimeOffset {
+    /// Left hand side (compared) [TimeScale]
+    lhs: TimeScale,
+    /// Right hand side (reference) [TimeScale]
+    rhs: TimeScale,
+    /// Weekly reference time (counter, nanoseconds)
+    t_ref_nanos: (u32, u64),
```

Comment:
This seems a bit too specific to GPST, no? Weekly reference and nanoseconds. Can we make this more generic somehow?

Reply:
Not it is totally generic, what makes you say that ?

It's just the calculations (the interpolation) uses time of week and week referencing, not timescale::t_0 referencing. 
And we have means to express weeks and reference to a certain week using your library

Reply:
I thought of GPST because I've never seen the start of week as something important in epochs outsides of GPST. Some astrodynamics data references day of year, but that's also pretty specific.